### PR TITLE
ci: Get rid of binaries submodule

### LINKED ISSUE #896: Move binaries from resources to external repo (or other place)
**Author the user story for this feature**

As a Developer, I want to move all binaries from `main/resources` to someplace to avoid using submodules (based on the research of task #849) so I can have a flank repository smaller.

**Is your feature request related to a problem? Please describe.**
External binaries should not be part of source code, to have quicker checkout and to keep repository size as small as possible.

**Describe the solution you'd like**
The solution should be chosen based on the research of task #849

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #896

This PR delete submodule for [binaries ](https://github.com/Flank/binaries) and download them at runtime from [release page ](https://github.com/Flank/binaries/releases) based on os system

`macOs` -> all should work by default because previously binaries were not copied on macOS
`linux` -> it download binaries from [release page](https://github.com/Flank/binaries/releases) at runtime to `<user_home>/.flank` directory and unpack them 
`windows` -> it works similar to `linux` version, however, full support will be available after #1134 

All tests should work properly

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Downloading binaries for Linux and Windows at runtime
- [x] Create binaries release for Linux and Windows
- [x] Update binaries assets after each change on binaries repo [release page](https://github.com/Flank/binaries/releases) >> [PR](https://github.com/Flank/binaries/pull/4)
- [x] Delete submodule
- [x] Update Github Actions to do not checkout submodules

### CODE REVIEW

File: buildSrc/src/main/kotlin/DownloadBinariesTask.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    @TaskAction
+    fun downloadBinaries() {
+        osName
+            .takeIf { forceUpdate || shouldDownloadBinaries() }
+            ?.let(::downloadAndUnzip)
+    }
+
+    private val osName: String
+        get() = if(isWindows) "windows" else "linux"
```
Comment (adamfilipow92) [inline buildSrc/src/main/kotlin/DownloadBinariesTask.kt:29]: Maybe add fail if mac?

Comment (piotradamczyk5) [reply buildSrc/src/main/kotlin/DownloadBinariesTask.kt:29]: Basically, it should contain these binaries for running flank.jar on Linux while building on ubuntu

Comment (piotradamczyk5) [reply buildSrc/src/main/kotlin/DownloadBinariesTask.kt:29]: Actually, I figure out that we should also have binaries for windows included. I will make logic to contains download both Windows and Linux when running the build on CI

**Review by jan-gogo — COMMENTED** [review]
The code looks well, but I would love to have this implementation in the java-library module linked to the test_runner as Gradle dependency, maybe not now, it could be in another PR. 
Some of the code looks a little bit redundant according to flank-scripts, for example, utilities for unzipping & symlinks, probably also could be moved to another library module shared between download-binaries and flank-scripts.

**Review by jan-gogo — APPROVED** [review]

**Comment by piotradamczyk5** [comment]
> The code looks well, but I would love to have this implementation in the java-library module linked to the `test_runner` as Gradle dependency, maybe not now, it could be in another PR.
> Some of the code looks a little bit redundant according to flank-scripts, for example, utilities for unzipping & symlinks, probably also could be moved to another library module shared between `download-binaries`(?) and `flank-scripts`.

Yeah I think that we could do some shared module which has common logic for other modules, I will create a ticket for it

File: docs/binaries_used_in_flank_ios_testing.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+They are unpacked to `<user directory>/.flank`.
+If they already exist on this path, they are not downloaded again.
+
+## Updating
+
+In order to update binaries just follow below steps:
+1. checkout binaries [repository](https://github.com/Flank/binaries)
+1. update them using:
+   - `updateBinariesWithFlankBash` will update binaries for Linux and Windows using `flank-scripts`
+   - `update.sh` (old method). It will update binaries for Linux OS
```
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/binaries_used_in_flank_ios_testing.md:19]: this is standalone without flank-scripts needed

**Review by adamfilipow92 — APPROVED** [review]
- [X] Tested on windows, works!

**Pull request merged**